client/src/components/Workflow/Editor/modules/model.ts

```diff
@@ -49,18 +54,32 @@ export async function fromSimple(
     // Galaxy assign new ones.
     if (options?.reassignIds ?? appendData) {
         const stepIdOffset = stepStore.getStepIndex + 1;
+        const stepArray = Object.values(data.steps);
 
-        Object.values(data.steps).forEach((step) => {
+        // Since we are resigning IDs based on index, ensure correct ordering
+        stepArray.sort((a, b) => a.id - b.id);
```

Comment:
Wouldn't you want to use the order_index for this ?

Reply:
I'm unsure. I used the id, so the new id's will be as close to possible as the ids from the source. What would the argument be for using order_index instead?

Reply:
It's sorted by the DAG, not when the step was created

Reply:
Ok. And how would that be better than using the old ids, for assigning the new ids?

Reply:
That's what the backend does anyway once you save your workflow. I guess it doesn't really matter.